cpp/src/neighbors/cagra.cuh

```diff
@@ -36,6 +38,36 @@
 
 namespace cuvs::neighbors::cagra {
 
+// Member function implementations for cagra::index
+template <typename T, typename IdxT>
+void index<T, IdxT>::compute_dataset_norms_(raft::resources const& res)
+{
+  // Get the dataset view
+  auto dataset_view = this->dataset();
+
+  // Allocate norms vector if not already allocated
+  if (!dataset_norms_.has_value() || dataset_norms_->extent(0) != dataset_view.extent(0)) {
+    dataset_norms_ = raft::make_device_vector<float, int64_t>(res, dataset_view.extent(0));
```

Comment:
If there is an existing vector, then let's deallocate that first. This could help when we work with very large dataset.
```suggestion
    dataset_norms_ = raft::make_device_vector<float, int64_t>(res, 0);
    dataset_norms_ = raft::make_device_vector<float, int64_t>(res, dataset_view.extent(0));
```